Delcam.ProductInterface.PowerSHAPE.Test/DatabaseEntity Tests/Entity Tests/MeshTest.cs

```diff
@@ -249,6 +250,52 @@ public void MinimumDistanceMeshCompCurve()
             //Assert.AreEqual(0.0, entity1.DistanceTo(entity2));
         }
 
+        /// <summary>
+        /// When writing mesh to a DMT check that mesh is kept the same: the same bounding box,
+        /// the same number of triangles
+        /// </summary>
+        [Test]
+        public void WhenWriteToDMTFile_CheckMeshDidNotChange()
+        {
+            // Import mesh
+            _powerSHAPE.Reset();
+            var importedMesh =_powerSHAPE.ActiveModel.Meshes.CreateMeshFromFile(new File(TestFiles.SINGLE_MESH));
+            var numberOfTriangles = importedMesh.NumberOfTriangles;
+            var boundingBox = importedMesh.BoundingBox;
+
+            // Export mesh 
+            var output = File.CreateTemporaryFile("dmt");
+            importedMesh.WriteToDMTFile(output);
+
+            // Test the distance
+            var exportedMesh = _powerSHAPE.ActiveModel.Meshes.CreateMeshFromFile(output);
+            Assert.AreEqual(numberOfTriangles, exportedMesh.NumberOfTriangles);
+            Assert.AreEqual(boundingBox, exportedMesh.BoundingBox);
+        }
+
+        /// <summary>
+        /// When writing mesh to a STL check that mesh is kept the same: the same bounding box,
+        /// the same number of triangles
+        /// </summary>
+        [Test]
+        public void WhenWriteToSTLFile_CheckMeshDidNotChange()
+        {
+            // Import mesh
+            _powerSHAPE.Reset();
+            var importedMesh = _powerSHAPE.ActiveModel.Meshes.CreateMeshFromFile(new File(TestFiles.SINGLE_MESH));
+            var numberOfTriangles = importedMesh.NumberOfTriangles;
+            var boundingBox = importedMesh.BoundingBox;
+
+            // Export mesh 
+            var output = File.CreateTemporaryFile("stl");
```

Comment:
You need to clean up this temporary file too.

Reply:
Done